src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestPolicyRepository.java

```diff
@@ -64,16 +72,43 @@ public CompletableFuture<Result<RequestPolicy>> lookupRequestPolicy(Item item, U
       .thenApply(result -> result.map(RequestPolicy::from));
   }
 
+  public CompletableFuture<Result<Collection<RequestPolicy>>> lookupRequestPolicies(
+    Collection<Item> items, User user) {
+
+    log.debug("lookupRequestPolicies:: parameters item: {}, user: {}",
+      () -> asJson(items), () -> asJson(user));
+
+    Set<CirculationRuleCriteria> criteriaSet = items.stream()
+      .map(item -> new CirculationRuleCriteria(item, user))
+      .collect(Collectors.toSet());
+    log.info("lookupRequestPolicies:: criteriaSet: {}", () -> asJson(criteriaSet));
+
+    return allOf(criteriaSet, criteria -> lookupRequestPolicyId(criteria.getMaterialTypeId(),
```

Comment:
This will run all requests in parallel. Should we try and run them sequentially?